### PR TITLE
add feature/b-1

### PR DESCRIPTION
**Pull request by OlehNaidiuk**

### CODE REVIEW

File: src/main/java/com/naidiuk/onlineshop/entity/Catalog.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.naidiuk.onlineshop.entity;
+
+public enum Catalog {
```
Comment (gurskiyy) [inline src/main/java/com/naidiuk/onlineshop/entity/Catalog.java]: i think, you should rename Catalog to CatalogType. It's more understandable. How do u think?

File: src/main/java/com/naidiuk/onlineshop/entity/Size.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    private Long sizeId;
+    @Column(name = "size_value")
+    private String value;
+    @ManyToMany
+    @JoinTable(
+            name = "category_size",
+            joinColumns = @JoinColumn(name = "category_id"),
+            inverseJoinColumns = @JoinColumn(name = "size_id")
+    )
+    private List<Category> categorySizes;
```
Comment (gurskiyy) [inline src/main/java/com/naidiuk/onlineshop/entity/Size.java]: u should rename categorySizes. Think about that. Search more examples about many to many and naming please

File: src/main/java/com/naidiuk/onlineshop/mapper/CategoryMapper.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.naidiuk.onlineshop.mapper;
+
+import com.naidiuk.onlineshop.dto.CategoryDto;
+import com.naidiuk.onlineshop.entity.Category;
+
+public class CategoryMapper {
```
Comment (gurskiyy) [inline src/main/java/com/naidiuk/onlineshop/mapper/CategoryMapper.java:6]: why don't u use builder in Lombok?

File: src/main/resources/SQLscript.txt [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+left join size
+on        size_id in (12, 13, 14, 15, 16, 17, 18, 19)
+where     category_id in (5, 6, 7);
+
+insert into category_size
+select    category_id, size_id
+from      category
+left join size
+on        size_id in (2, 3, 4, 5, 12, 14, 16)
+where     category_id in (8);
```
Comment (gurskiyy) [inline src/main/resources/SQLscript.txt]: 1. this file should format .sql
2. this file doesn't used in project and doesn't invoke in project for now. 
- so for now, i propose you, create directory on project root some like that db_scripts and put them there

File: src/test/java/com/naidiuk/onlineshop/integration/repository/CategoryRepositoryTest.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    private Category createCategory() {
+        Category category = new Category();
+        category.setCategoryId(1L);
+        category.setName("кеди");
+        category.setCatalog(Catalog.FOOTWEAR);
+        return category;
+    }
+}
```
Comment (gurskiyy) [inline src/test/java/com/naidiuk/onlineshop/integration/repository/CategoryRepositoryTest.java:28]: this test not really do. u need test findAll to your own DB. and that all. then u will use some another library for decide problems

File: src/test/java/com/naidiuk/onlineshop/modular/controller/CategoryControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    @Mock
+    private CategoryService mockedCategoryService;
+    @InjectMocks
+    private CategoryController categoryController;
+
+    @Test
+    void findAll() {
+        //prepare
+        List<CategoryDto> categoriesDto = new ArrayList<>();
+        CategoryDto firstDto = mock(CategoryDto.class);
```
Comment (gurskiyy) [inline src/test/java/com/naidiuk/onlineshop/modular/controller/CategoryControllerTest.java:33]: pls swap 33 and 32 lines

File: src/main/java/com/naidiuk/onlineshop/entity/Catalog.java (hunk 1 above)
Comment (gurskiyy) [inline src/main/java/com/naidiuk/onlineshop/entity/Catalog.java]: lombok?

**Comment by OlehNaidiuk** [comment]
dfgdfsgdf

**Pull request merged**